packages/python-packages/apistubgentest/apistubgentest/models/_models.py

```diff
@@ -149,7 +167,54 @@ def __init__(self, name: str = "Bob", age: int = 21, is_awesome: bool = True, pe
         self.is_awesome = is_awesome
         self.pet = pet
 
+
 class SomePoorlyNamedObject:
 
     def __init__(self, name: str):
         self.name = name
+
+
+class SomethingWithOverloads:
+
+    @overload
+    def double(self, input: int = 1, *, test: bool = False, **kwargs) -> int:
+        ...
+
+    @overload
+    def double(self, input: Sequence[int] = [1], *, test: bool = False, **kwargs) -> list[int]:
```

Comment:
Don't the overloads need to be `async def` as well?

Reply:
Well, I was following the example Laurent reported here: https://github.com/Azure/azure-sdk-tools/issues/3012

Reply:
That.... doesn't sound right....
I think the overloads in ACR might need fixing?

Reply:
If that is infact a bug in ACR (still TBD) - then that might be a good thing to have a linter rule for...

Reply:
I guess based on the rules that @swathipil was discussing about how mypy uses `@decorator` it doesn't really make sense to have sync overloads with an async implementation, though from APIView's perspective it's purely syntactical.

Reply:
I created an issue for this: https://github.com/Azure/azure-sdk-for-python/issues/23907
Depending on how this is written, mypy may or may not complain, but this is not a supported case. I'll update the test for just sync.

Reply:
Test package updated. This should be flagged as an inappropriate use over @overload in ACR.

Reply:
I will open an issue against ACR